packages/angular_devkit/build_facade/src/build-system.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Path, asPosixPath, asWindowsPath, dirname, logging, schema } from '@angular-devkit/core';
+import { readFileSync } from 'fs';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { _throw } from 'rxjs/observable/throw';
+import { mergeMap } from 'rxjs/operators';
+import { TargetOptions } from './architect';
+
+
+export interface BuildContext {
+  logger: logging.Logger;
+}
+
+export interface BuildStats {
+  start: Date;
+  end: Date;
+  hash: string;
+
+  totalSize: number;
+  criticalSize: number;
+  initialSize: number;
+  assetsSize: number;
+}
+
+export interface BuildAsset {
+  // TODO: figure out virtual file system host
+  // host: Host;
+  name: string;
+  path: Path;
+  size: number;
+  flags: string[];
+}
+
+export interface BuildArtifact {
+  name: string;
+  outputs: BuildAsset[];
+  flags: string[];
+}
+
+export interface BuildEvent {
+  success: boolean;
+  artifacts: BuildArtifact[];
+  stats: BuildStats;
+}
+
+// TODO: need a way to show/return option help.
+export class Target {
+  constructor(public schemaPath: Path) { }
+
+  protected validateOptions(options: TargetOptions): Observable<TargetOptions> {
+    const schemaContent = readFileSync(this._getSystemPath(this.schemaPath), 'utf-8');
+    const workspaceSchema = JSON.parse(schemaContent);
+    const registry = new schema.CoreSchemaRegistry([],
+      this._getSystemPath(dirname(this.schemaPath)));
+
+    return registry.compile(workspaceSchema).pipe(
+      mergeMap(validator => validator(options)),
+      mergeMap(validatorResult => {
+        if (validatorResult.success) {
+          return of(options);
+        } else {
+          return _throw(validatorResult.errors);
```

Comment:
`errors` is an array of strings. You should create a new error object since that's what's expected.